[WIP] Implemented the clean up methods.

phillipsj opened the pull request:
I am sure that I may have missed something, I just wanted to get this up as a place to start discussing and making improvements.

luthermonson commented on pkg/agent/clean/node_windows.go:
```diff
@@ -1,35 +1,224 @@
 func Links() error {
-	// TODO
+	wins := createWinsCommand("Network")
+	cmd := exec.Command("wins.exe", wins)
```
Can just return exec.Command(...).Run() on all of theses

luthermonson commented on pkg/agent/clean/node_windows.go:
```diff
@@ -1,35 +1,224 @@
 	return nil
 }
+
+func createWinsCommand(cmd string) string {
```
This actually returns the args, either return an exec.Command or call it createWinsArgs

luthermonson commented on pkg/agent/clean/node_windows.go:
```diff
@@ -1,35 +1,224 @@
 }
+
+func createWinsCommand(cmd string) string {
+	pathPrefix := getPrefixPath()
```
No need to set a new ver if you use it once, just put getPrefixPath() in the sprintf below

luthermonson replied:
Correction, the join below

luthermonson commented on pkg/agent/clean/node_windows.go:
```diff
@@ -1,35 +1,224 @@
+
+func createWinsCommand(cmd string) string {
+	pathPrefix := getPrefixPath()
+	script := path.Join(pathPrefix, "etc", "rancher", "cleanup.ps1")
```
Seems dumb but Command takes a bunch of strings as args and it's better imho to return back a []string of each arg separate and do a ... on the return than pass one string as the second arg to Command

phillipsj committed:
Implemented the clean up methods.

Updated the PowerShell script and leverages the wins cli to execute the PowerShell functions.



Feedback changes.
